rdrf/rdrf/templates/rdrf_cdes/form_list.html

```diff
@@ -0,0 +1,101 @@
+{% extends "rdrf_cdes/base-1-col.html" %}
+
+{% load add_attr %}
+{% load static from staticfiles %}
+{% load get_patient %}
+{% load has_feature %}
+{% load get_form_element %}
+{% load is_patient_relative_widget %}
+{% load is_formset_obj %}
+{% load get_information_link %}
+{% load get_info_text_expression %}
+{% load i18n %}
+{% load translate %}
+
+
+{% block formlinks %}
+{% endblock %}
+
+{% block sidebar %}
+    List view
+{% endblock %}
+
+{% block formbtns %}
+{% endblock %}
+
+{% block extrahead %}
+    <script type="text/javascript">
+        var formLinks = JSON.parse("{{form_links|escapejs}}");
+
+        function listLinks(query) {
+            var linkList = document.getElementById("link-list");
+            linkList.innerHTML = "";
+
+            formLinks.forEach(function (linkItem) {
+                if (!linkItem["text"].includes(query)) {
+                    return;
+                }
+
```

Comment:
This keeps re-creating the DOM elements on each keystroke.
Given that we start by listing all the links and therefore creating all of them, can't we just move the creation of all the form links up to line 29 and do it just once?

Reply:
I'm not really sure what you mean. The idea is that the links are only created and thus visible when they match the query. Are you suggesting that we create all the links initially, and then delete them if they don't match?

We could also use timeouts to reduce the number of useless cycles by waiting a little after each keystroke before performing the query.

Reply:
Also, we could use ES5's `filter` but I'm not sure what JS version we're targeting

Reply:
Yes, I'm suggesting that you create all the links just once on initialisation.
You would never delete them after that, just keep them around.
When the query changes you clear all the existing links, then you attach the ones that match the query.

You can use ``filter`` form Underscore.js (that is included) or ``grep`` from jQuery instead of relying on ES5's ``filter``.

Reply:
Ah, gotcha. That makes sense.